modules/ppcp-applepay/src/Assets/ApplePayDataObjectHttp.php

```diff
@@ -217,17 +217,11 @@ public function update_method_data(): void {
 	 * @param string $caller_page The caller page.
 	 */
 	public function order_data( string $caller_page ): void {
-		$nonce = filter_input( INPUT_POST, 'woocommerce-process-checkout-nonce', FILTER_SANITIZE_SPECIAL_CHARS );
-		if ( ! $nonce ) {
-			return;
-		}
-		$is_nonce_valid = wp_verify_nonce(
-			$nonce,
-			'woocommerce-process_checkout'
-		);
-		if ( ! $is_nonce_valid ) {
+
+		if ( ! $this->is_nonce_valid() ) {
 			return;
 		}
+		//phpcs:disable WordPress.Security.NonceVerification
 		$data = filter_var_array( $_POST, FILTER_SANITIZE_SPECIAL_CHARS );
```

Comment:
It's ok, no need to change, but in this plugin the mostly used method is to send the request body as JSON [https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-googlepay/resources/js/Helper/UpdatePaymentData.js#L18](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-googlepay/resources/js/Helper/UpdatePaymentData.js#L18) then use `->read_request()` which already handles the nonce [https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-googlepay/src/Endpoint/UpdatePaymentDataEndpoint.php#L70](https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/modules/ppcp-googlepay/src/Endpoint/UpdatePaymentDataEndpoint.php#L70)